actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -231,6 +231,9 @@ class Parameters
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
     def initialize(parameters = {})
+      unless parameters.respond_to?(:with_indifferent_access)
```

Comment:
In the case where an actual error occurs, rescuing is more expensive. (`rescue` is essentially free, `raise` is the expensive bit). `respond_to?` isn't the most costly thing in the world, but it's non-zero cost. Basically doing `rescue` means that if no error occurs, we are doing no additional work.